### ISSUE TITLE
[BUG] Error during PDF creation during OspreyFit outside GUI

### ISSUE THREAD

**Issue by nwd2918** [schorschinho/osprey#244]
**Describe the bug**

When running the `OspreyFit` function from the terminal with the 'savePDF` flag set the following errors result when figures are being created:

### Error 1
```
Reference to non-existent field 'diff1'.

Error in osp_plotModule (line 333)
                        ' ppm\nNumber of metabolites: '
                        num2str(MRSCont.fit.resBasisSet.(which){1,MRSCont.info.A.unique_ndatapoint_indsort(kk)}.nMets)
                        '; Number of macro moclecules: '
                        num2str(MRSCont.fit.resBasisSet.(which){1,MRSCont.info.A.unique_ndatapoint_indsort(kk)}.nMM)
                        ...

Error in osp_plotAllPDF (line 86)
                    osp_plotModule(MRSCont, 'OspreyFit', kk, Names{ss});

Error in OspreyFit (line 215)
    osp_plotAllPDF(MRSCont, 'OspreyFit')
```

Modifying the relevant line (333) in `ops_plotModule.m` from 
```
' ppm\nNumber of metabolites: ' num2str(MRSCont.fit.resBasisSet.(which){1,MRSCont.info.A.unique_ndatapoint_indsort(kk)}.nMets) '; Number of macro moclecules: ' num2str(MRSCont.fit.resBasisSet.(which){1,MRSCont.info.A.unique_ndatapoint_indsort(kk)}.nMM) ...
```
to 
```
 ' ppm\nNumber of metabolites: ' num2str(MRSCont.fit.resBasisSet.conc{1,MRSCont.info.A.unique_ndatapoint_indsort(kk)}.nMets) '; Number of macro moclecules: ' num2str(MRSCont.fit.resBasisSet.conc{1,MRSCont.info.A.unique_ndatapoint_indsort(kk)}.nMM) ...
```
appears to resolve the issue and I think retains the intended function.

### Error 2
```
Not enough input arguments.

Error in osp_plotFit (line 56)
            figTitle = sprintf([fitMethod ' ' fitStyle ' ' conc ' fit plot:\n' filen
            ext]);

Error in osp_plotModule (line 419)
            temp = osp_plotFit(MRSCont, kk,Style,which);

Error in osp_plotAllPDF (line 86)
                    osp_plotModule(MRSCont, 'OspreyFit', kk, Names{ss});

Error in OspreyFit (line 215)
    osp_plotAllPDF(MRSCont, 'OspreyFit')
```
Modifying the relevant line in `osp_plotFit.m` from 
```
figTitle = sprintf([fitMethod ' ' fitStyle ' ' conc ' fit plot:\n' filen ext]);
```
to 
```
figTitle = sprintf([fitMethod ' ' fitStyle ]);
```
resolves the issue. It's not clear if any functionality is lost. 

**Computer configuration (please complete the following information):**
 - OS: Ubuntu 18.04
 - MATLAB version: 2020b

**Comment by HJZollner** [comment]
Hi!

Thanks for pointing out this issue. We have fixed it in #245.

Apparently this bug is related to the 'Concatenated' fit style for edited data only. I'd advise you to not use the 'Concatenated' fitting, as it appears to be unreliable in its current implementation. Therefore, we did not test the plot outcome of this model recently.

Best,
Helge

**Comment by nwd2918** [comment]
That's useful to know, thank you.

Niall
